SCF-Session5-Question2/src/meta/utility/BinarySearch.java

```diff
@@ -0,0 +1,55 @@
+/*
+* This Class contains serach function 
+* using binary search algorithm
+* @author ABHISHEK BISHNOI
+* @version 1.0
+* @since 17-July-2019
+*/
+package meta.utility;
+
+public class BinarySearch implements Search {
+	int firstIndex=-1;
+	int lastIndex=-1;
+	/*
+	 * (non-Javadoc)
+	 * @see meta.utility.Search#search(int, int[])
+	 * @param int value to be search
+	 * @param int[] dataset where we search
+	 * @return int index of that element
+	 * Assumption All Values Are Positive
+	 * Assumption Array is always in sorter order
+	 * Assumption Values are Not repeat into array data
+	 * Exception AssertionError when array size is zero
+	 */
+	@Override
+	public int search(int value, int[] array) {
```

Comment:
In case of mutually exclusive if statements, using switch case would be a better option.